File: leetcode/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if len(intervals) == 1:
            return intervals

        ans = []
        m = 0
        temp = []
        intervals.sort()
        for i in range(len(intervals)*2):
            if len(temp) == 0:
                temp = intervals[0]
                intervals.pop(0)
                continue

            one, two = temp
            if intervals:
                th, fo = intervals[0]
                if two >= th and two <= fo:
                    temp = [min(one, th), max(fo, two)]
                    intervals.pop(0)
                elif fo <= two and th >= one:
                    temp = [min(one, th), max(fo, two)]
                    intervals.pop(0)
                else:
                    ans.append(temp)
                    temp = []
        if temp:
            ans.append(temp)
        return ans

File: leetcode/merge_intervals.go
import "sort"

func merge(intervals [][]int) [][]int {
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})
	ans := make([][]int, 0)
	temp := intervals[0]
	for i := 1; i < len(intervals); i++ {
		if temp[0] <= intervals[i][0] && intervals[i][0] <= temp[1] {
			if temp[1] < intervals[i][1] {
				temp[1] = intervals[i][1]
			}
		} else {
			ans = append(ans, temp)
			temp = intervals[i]
		}
	}
	ans = append(ans, temp)
	return ans
}